synchronizer/synchronizer.go

```diff
@@ -408,19 +409,19 @@ func (s *ClientSynchronizer) Sync() error {
 				continue
 			}
 			log.Infof("latestSequencedBatchNumber: %d, latestSyncedBatch: %d, lastVerifiedBatchNumber: %d", latestSequencedBatchNumber, latestSyncedBatch, lastVerifiedBatchNumber)
-
+			resetDone := false
 			// Sync trusted state
 			// latestSyncedBatch -> Last batch on DB
 			// latestSequencedBatchNumber -> last batch on SMC
 			if latestSyncedBatch >= latestSequencedBatchNumber {
 				startTrusted := time.Now()
 				if s.syncTrustedStateExecutor != nil && !s.isTrustedSequencer {
 					log.Info("Syncing trusted state (permissionless)")
-					if s.asyncL1BlockChecker != nil {
-						if s.asyncL1BlockChecker.OnStartL2Sync(s.ctx) {
-							log.Warn("s.asyncL1BlockChecker.OnStartL2Sync -> continue")
-							continue
-						}
+					//Sync Trusted State
+					resetDone, lastEthBlockSynced, err = s.checkReorgAndExecuteReset(lastEthBlockSynced)
+					if resetDone || err != nil {
+						log.Infof("Reset done before L2 sync")
+						continue
 					}
```

Comment:
I think err should be handled. At least the error must be printed even if it is not returned
```suggestion
					if resetDone {
						log.Infof("Reset done before L2 sync")
						continue
					} else if || err != nil {
					       log.Error("error in checkReorgAndExecuteReset function. Error: ", err)
					       return err
```